cypress/tests/ui/flowchart/flowchart.cy.js

```diff
@@ -1 +1,166 @@
+      'nodeToToggle'
+    );
+
+    // Assert before action
+    cy.get('@nodeToToggle').should('not.be.checked');
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('@nodeToToggle').check({ force: true });
+    cy.get('.pipeline-node--parameters').first().click({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open the metadata panel for datasets. #TC-27', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--data').first().click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+  });
+
+  it('verifies that users can open and preview the dataset table in the metadata panel for datasets. #TC-28', () => {
+    // Assert before action
+    cy.get('.pipeline-metadata-modal').should('not.exist');
+
+    // Action
+    cy.get('.pipeline-node.pipeline-node--data').first().click();
+    cy.get('.pipeline-metadata__link').click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata-modal').should('exist');
+    cy.get('.pipeline-metadata-modal__preview').should('exist');
+    cy.get('.preview-table').should('exist');
+  });
+
+  it('verifies that users can open the metadata panel showing the Plotly/Matplotlib preview. #TC-29', () => {
+    const nodeToClickText = 'Price Histogram';
+
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+    cy.get('.plot-container.plotly').should('not.exist');
+
+    // Action
+    cy.contains('text', nodeToClickText).click({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+    cy.get('.plot-container.plotly').should('exist');
+  });
+
+  it('verifies that users can open and preview the Plotly/Matplotlib charts in the metadata panel. #TC-30', () => {
+    const nodeToClickText = 'Price Histogram';
+
+    // Assert before action
+    cy.get('.pipeline-metadata-modal').should('not.exist');
+
+    // Action
+    cy.contains('text', nodeToClickText).click({ force: true });
+    cy.get('.pipeline-metadata__link').click();
+
+    // Assert after action
+    cy.get('.pipeline-metadata-modal').should('exist');
+    cy.get('.pipeline-plotly-chart.pipeline-plotly__modal').should('exist');
+  });
+
+  it('verifies that users can open the metadata panel showing the R2 score details. #TC-31', () => {
+    const nodeToClickText = 'R2 Score';
+
+    // Assert before action
+    cy.get('.pipeline-metadata--visible').should('not.exist');
+
+    // Action
+    cy.contains('text', nodeToClickText).click({ force: true });
+
+    // Assert after action
+    cy.get('.pipeline-metadata--visible').should('exist');
+    cy.contains('.pipeline-metadata__title', nodeToClickText);
```

Comment:
maybe the assertion should be if tracking data from last run is visible (actually the json tree div component)